src/microbe_stage/SpawnSystem.cs

```diff
@@ -8,6 +8,11 @@
 /// </summary>
 public class SpawnSystem
 {
+    /// <summary>
+    ///   Estimate count of existing spawned entities grouped by the spawner
+    /// </summary>
+    private readonly Dictionary<Type, int> entityCountPerSpawner = new Dictionary<Type, int>();
```

Comment:
I think that instead of mapping from type, you should associate these with the exact Spawner objects, that way don't need to add a bunch of SpawnerType things around, and it'll nicely work to make sure that the ratios between species is correct. So for example a species that has a much higher population would also translate to it having a bigger allowance for number of active cells.

Reply:
The problem with having a `Dicrionary<Spawner, int>` is that I would need a possibillity to acquire the Spawner from a ISpawned object.

Reply:
Why? Just have the lists be specific to a Spawner. That way if ISpawned is in a list, you always know which Spawner it is related to.

Reply:
Do you mean I should add a `List<ISpawned>` to `Spawner`?

Reply:
Yes. And optionally remove the other places where there are ISpawned list / maybe it is time to even remove the spawned Node tag even, to reduce needing to do any kind of mapping between the different lists.

Reply:
This sounds like a good idea, but there is one problem: the `DespawnEntities` method does not hold a reference to a Spawner, it only has a reference to ISpawned. I could iterate through all Spawners and check if the entity to be removed is in the list.

Reply:
I didn't look but could DespawnEntities be changed to take in List<Spawner> and then it can iterate the lists in the spawners themselves? That way the Spawner would be the authoritative place for all spawned entity lists.